providers/zoho.go

```diff
@@ -0,0 +1,82 @@
+package providers
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/pusher/oauth2_proxy/pkg/apis/sessions"
+	"github.com/pusher/oauth2_proxy/pkg/requests"
+)
+
+// ZohoProvider represents a Zoho based Identity Provider
+type ZohoProvider struct {
+	*ProviderData
+}
+
+// NewZohoProvider initiates a new ZohoProvider
+func NewZohoProvider(p *ProviderData) *ZohoProvider {
+	p.ProviderName = "Zoho"
+	if p.LoginURL.String() == "" {
+		p.LoginURL = &url.URL{Scheme: "https",
+			Host:     "accounts.zoho.com",
+			Path:     "/oauth/v2/auth",
+			RawQuery: "response_type=code&access_type=offline",
+		}
+	}
+	if p.RedeemURL.String() == "" {
+		p.RedeemURL = &url.URL{Scheme: "https",
+			Host: "accounts.zoho.com",
+			Path: "/oauth/v2/token",
+		}
+	}
+	if p.ProfileURL.String() == "" {
+		p.ProfileURL = &url.URL{Scheme: "https",
+			Host: "accounts.zoho.com",
+			Path: "/oauth/user/info",
+		}
+	}
+	if p.ValidateURL.String() == "" {
+		p.ValidateURL = p.ProfileURL
+	}
+	if p.Scope == "" {
+		p.Scope = "ZohoProfile.userinfo.read,Aaaserver.profile.read"
+	}
+	return &ZohoProvider{ProviderData: p}
+}
+
+func getZohoHeader(accessToken string) http.Header {
+	header := make(http.Header)
+	header.Set("Content-Type", "application/json")
+	header.Set("Authorization", fmt.Sprintf("Bearer %s", accessToken))
+	return header
+}
```

Comment:
This is identical to [getOIDCToken](https://github.com/pusher/oauth2_proxy/blob/master/providers/oidc.go#L175-L180), perhaps that could be re-used?

Reply:
@JoelSpeed the response type is indeed duplicated, i will fix. The access type needs to be sent for zoho. Online means only access token is sent and lasts for 1 hour. Offline, they send a refresh token.

About reusing methods directly from different strategies, IMHO its not a good ideia, but i can change it if you want. 

Maybe we can create an abstract oauth provider that can be flexible to setup from config files.

Reply:
> Offline, they send a refresh token.

This is common to many providers, I wonder why this isn't being set anywhere else 🤔 

> About reusing methods directly from different strategies, IMHO its not a good ideia, but i can change it if you want.

Happy to leave it, thanks for pushing back :) 

> Maybe we can create an abstract oauth provider that can be flexible to setup from config files.

Yes x1000!!! I have been thinking about this for months, trying to work out a good way to do this and transition the existing providers is a problem for me though